In [29]:
from math import factorial as f
import numpy as np

In [30]:
def wer(a,b):
    return f(a) / (f(b) * f(a-b))
def fbernull(c, a, b):
    return wer(a,b) * c**(b) * (1-c)**(a-b)
def puason(n,m):
    return n**m / f(m) * np.exp(-n)


1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [31]:
fbernull(0.8, 100, 85)

0.048061793700746355

0.048061793700746355

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [32]:
a = 5000 * 0.0004
puason(a , 0)

0.1353352832366127

In [33]:
puason(a , 2)

0.2706705664732254

0.1353352832366127     0.2706705664732254

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [34]:
fbernull(0.5, 144, 70 )

0.06281178035144776

0.06281178035144776

4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

Какова вероятность того, что все мячи белые?

In [35]:
(7*6*9*8)/(10*9*11*10)

0.3054545454545455

0.3054545454545455

Какова вероятность того, что ровно два мяча белые? 

In [36]:
(7/10)*(6/9)*(2/11)*(1/10) + (7/10)*(3/9)*(9/11)*(2/10) + (7/10)*(3/9)*(2/11)*(9/10) + \
(3/10)*(7/9)*(9/11)*(2/10) + (3/10)*(7/9)*(2/11)*(9/10) + (3/10)*(2/9)*(9/11)*(8/10)


0.20484848484848484

0.20484848484848484

 Какова вероятность того, что хотя бы один мяч белый?

In [37]:
1-(3/10*2/9*2/11*1/10)

0.9987878787878788

0.9987878787878788